src/Database/LSMTree/Internal.hs

```diff
@@ -1100,13 +1104,34 @@ open sesh label snap = do
               FS.ReadMode $ \h ->
                 FS.hGetAll (sessionHasFS seshEnv) h
       let (label', runNumbers, conf) = read . BSC.unpack . BSC.toStrict $ bs
+      let conf' = applyOverride override conf
       unless (label == label') $ throwIO (ErrSnapshotWrongType snap)
       let runPaths = V.map (bimap (second $ RunFsPaths (Paths.activeDir $ sessionRoot seshEnv))
                                   (V.map (RunFsPaths (Paths.activeDir $ sessionRoot seshEnv))))
                            runNumbers
-      with (openLevels hfs hbio (confDiskCachePolicy conf) runPaths) $ \lvls -> do
+      with (openLevels hfs hbio (confDiskCachePolicy conf') runPaths) $ \lvls -> do
         am <- newArenaManager
-        (newWith sesh seshEnv conf am WB.empty lvls)
+        (newWith sesh seshEnv conf' am WB.empty lvls)
+
+data TableConfigOverride = TableConfigOverride {
+      -- | Override for 'confDiskCachePolicy'
+      confOverrideDiskCachePolicy  :: Maybe DiskCachePolicy
+    }
+
+applyOverride :: TableConfigOverride -> TableConfig -> TableConfig
+applyOverride TableConfigOverride{..} conf = conf {
+      confDiskCachePolicy = fromMaybe (confDiskCachePolicy conf) confOverrideDiskCachePolicy
+    }
+
+configNoOverride :: TableConfigOverride
+configNoOverride = TableConfigOverride {
+      confOverrideDiskCachePolicy = Nothing
+    }
+
+configOverrideDiskCachePolicy :: DiskCachePolicy -> TableConfigOverride
+configOverrideDiskCachePolicy pol = TableConfigOverride {
+      confOverrideDiskCachePolicy = Just pol
+    }
```

Comment:
This does not compose, there is no way to create an override that overrides two settings. Maybe add a monoid instance using `Last` for the fields? This would also give you a convenient way of specifying no overrides.

Reply:
I've added it in